NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -19,40 +24,544 @@
+        The declination to use for the magnetic field, in internal units
+    site: str, default=None
+        Instead of declination, a site name can be given to retrieve the declination using the magnetic field
+        as obtained from the radiotools.helper.get_magnetic_field_vector() function
+
+    Returns
+    -------
+    evt: NuRadioReco.framework.event.Event
+        Event object containing the CORSIKA information
+    """
+    if declination is None:
+        if site is not None:
+            try:
+                magnet = hp.get_magnetic_field_vector(site)
+                declination = hp.get_declination(magnet)
+                logger.info(
+                    f"Declination obtained from site information, is set to {declination / units.degree:.2f} deg"
+                )
+            except KeyError:
+                declination = 0
+                logger.warning(
+                    "Site is not recognised by radiotools. Defaulting to 0 degrees declination. "
+                    "This might lead to incorrect electric field polarizations."
+                )
+        else:
+            declination = 0
+            logger.warning(
+                "No declination or site given, assuming 0 degrees. "
+                "This might need to incorrect electric field polarizations."
+            )
+
+    corsika = h5py.File(input_file, "r")
+
+    sampling_rate = 1. / (corsika['CoREAS'].attrs['TimeResolution'] * units.second)
+    zenith, azimuth, magnetic_field_vector = get_angles(corsika, declination)
+
+    # The traces are stored in a SimStation
+    sim_station = NuRadioReco.framework.sim_station.SimStation(0)  # set sim station id to 0
+
+    sim_station.set_is_cosmic_ray()
+
+    for j_obs, observer in enumerate(corsika['CoREAS']['observers'].values()):
+        obs_positions_geo = convert_obs_positions_to_nuradio_on_ground(
+            observer.attrs['position'], zenith, azimuth, magnetic_field_vector
+        )
+        efield, efield_time = convert_obs_to_nuradio_efield(
+            observer, zenith, azimuth, magnetic_field_vector
+        )
+
+        add_electric_field_to_sim_station(
+            sim_station, [j_obs],
+            efield, efield_time[0],
+            zenith, azimuth,
+            sampling_rate,
+            efield_position=obs_positions_geo
+        )
+
+    evt = NuRadioReco.framework.event.Event(corsika['inputs'].attrs['RUNNR'], corsika['inputs'].attrs['EVTNR'])
+    evt.set_event_time(corsika['CoREAS'].attrs['GPSSecs'], format="gps")
+
+    stn = NuRadioReco.framework.station.Station(0)  # set station id to 0
+    stn.set_sim_station(sim_station)
+    evt.set_station(stn)
+
+    sim_shower = hdf5_sim_shower(corsika)
+    evt.add_sim_shower(sim_shower)
+
+    corsika.close()
+
+    return evt
+
+
+def hdf5_sim_shower(corsika, declination=0):
+    """
+    Creates an NuRadioReco `RadioShower` from a CoREAS HDF5 file, which contains the simulation inputs shower parameters.
```

Comment:
I think hdf5_sim_shower and create_sim_shower are too similar. What do you think about create_radio_shower?